src/workerd/api/tests/strict-compression-test.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2024 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+import { rejects } from 'node:assert';
+
+// Test strictCompression flag: closing without finishing decompression should error
+export const strictCompressionCloseWithoutFinish = {
```

Comment:
Non-blocking nit: It'd probably make sense to move these to compression-streams-test -- they were all in the same file internally. But it doesn't matter much.